python/pyspark/sql/functions.py

```diff
@@ -14102,7 +14102,7 @@ def sort_array(col: "ColumnOrName", asc: bool = True) -> Column:
     Collection function: sorts the input array in ascending or descending order according
     to the natural ordering of the array elements. Null elements will be placed at the beginning
     of the returned array in ascending order or at the end of the returned array in descending
-    order.
+    order. The natural order of a struct is the natural order of the first field in the struct schema.
```

Comment:
Oh, so it doesn't sort by the second field?

Reply:
Maybe you are right and StackOverflow is wrong. I'm not so good at Scala, but I think it is decided [here ](https://github.com/apache/spark/blob/398bff77b1c837aed55f4f10ff1157f7da813570/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateOrdering.scala#L58) or perhaps [here](https://github.com/apache/spark/blob/398bff77b1c837aed55f4f10ff1157f7da813570/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/types/PhysicalDataType.scala#L318)

Maybe it is sorting first on the first field, then on second and so on. 

A more correct description would in this case be 
```suggestion
    order. The natural order of a struct is the natural order of the underlying fields of the struct schema in positional order.
```

I'm not 100 % sure this is true either. Are we sure that we get the field in positional order when it iterates through the fields?